### DISCUSSION TITLE
BHT-002-GCLZB SetPoint

### DISCUSSION THREAD

**Discussion by Minims** [Koenkk/zigbee2mqtt#6453] [Devices]
I have a BHT-002-GCLZB and Zigbee2MQTT allow me to update SetPoint by 1°C.
On the device, step is by 0.5°C. 
Is it possible to update the step to set it by 0.5°C as it does with the Up/Down button ?

Currently trying to push 19.5 to MQTT make the climate device reset to 0°C.

Thx!

**Comment by Minims** [comment]
It seems too that the tuya gateway allow to update some value such as ‘temperature commensation’ can we implement that in zigbee2mqtt ?

**Reply by Minims** [reply]
@Koenkk Any idea about the configuration of ‘temperature compensation’ via Zigbee on BHT-002-GCLZB ? Many Thanks.

**Reply by Koenkk** [reply]
I don't know this device, hopefully somebody else can help

**Reply by Minims** [reply]
No problem, thanks for your time and this awesome project.

**Comment by presslab-us** [comment]
It looks like it is coded for 1 deg increments.  You can try changing this line:
https://github.com/Koenkk/zigbee-herdsman-converters/blob/ecb233fd10ac558ee6cbf5fc2961dc5aa0dd143f/devices.js#L14544
like so:
`        exposes: [e.child_lock(), exposes.climate().withSetpoint('current_heating_setpoint', 5, 30, 0.5, ea.STATE_SET)`

For temperature calibration you can try changing this line:
https://github.com/Koenkk/zigbee-herdsman-converters/blob/ecb233fd10ac558ee6cbf5fc2961dc5aa0dd143f/devices.js#L14545
like so:
`            .withLocalTemperature(ea.STATE).withLocalTemperatureCalibration()`

**Reply by Minims** [reply]
> https://github.com/Koenkk/zigbee-herdsman-converters/blob/ecb233fd10ac558ee6cbf5fc2961dc5aa0dd143f/devices.js#L14544
> like so:
> ` exposes: [e.child_lock(), exposes.climate().withSetpoint('current_heating_setpoint', 5, 30, 0.5, ea.STATE_SET)`

it seems keeping 1 degree step in the interface. pushing 20.5 ou 20,5 make set point reset to 0°C.

> For temperature calibration you can try changing this line:
> https://github.com/Koenkk/zigbee-herdsman-converters/blob/ecb233fd10ac558ee6cbf5fc2961dc5aa0dd143f/devices.js#L14545
> like so:
> ` .withLocalTemperature(ea.STATE).withLocalTemperatureCalibration()`

Thanks @presslab-us , this is working great !! It was my major issue with this Thermostat.
Do I need to make a PR for this ? 

I have few other questions. 
It seems this thermostat manage : 

Programs : 4 programs / week, 4 programs Saturday, 4 programs Sunday
Economy Mode and Economy Temp
Min Set temp, Max Set temp
Low / High Temperature Protection
Deadzone Temperature (don't know what is this).

Looking a way to integrate all of this in zigbee2mqtt.

Last information. the temperature in report in x2 . 40°C is 20 and 34°C is 17.
`
Zigbee2MQTT:info  2021-03-01 22:34:17: MQTT publish: topic 'zigbee2mqtt/Thermostat', payload '{"child_lock":"UNLOCKED","current_heating_setpoint":13,"heat":"OFF","linkquality":110,"local_temperature":25,"local_temperature_calibration":0,"min_temperature":29,"preset":"hold","preset_mode":"hold","program":[{"p1":"6h:45m 40°C"},{"p2":"8h:15m 34°C"},{"p3":"12h:0m 40°C"},{"p4":"14h:0m 40°C"},{"sa1":"8h:0m 40°C"},{"sa2":"12h:0m 40°C"},{"sa3":"14h:0m 40°C"},{"sa4":"18h:0m 40°C"},{"su1":"8h:0m 40°C"},{"su2":"12h:0m 40°C"},{"su3":"14h:0m 40°C"},{"su4":"18h:0m 40°C"}],"system_mode":"heat"}'
`

Thanks for the help

**Reply by presslab-us** [reply]
I'm sure a PR would be appreciated!

For the schedule, look at the other devices supporting `tz.tuya_thermostat_weekly_schedule`, maybe you can copy some things over.

**Reply by kds69** [reply]
Any volonteer for a PR?

**Reply by kasi0815** [reply]
I tried the suggested changes above. local_temperature_calibration is now included in the latest release but setting current_heating_setpoint to 20.5 still reset the value to 0°C! Does anybody has an idea how to fix it? 
Thanks in advance   :-)

**Comment by bwims** [comment]
Bump.

I don't have the skills to modify this, but I would dearly like to calibrate by half degrees! The display is always half a degree wrong!

**Reply by Minims** [reply]
Not possible, this zigbee device only manage 1°C step via zigbee.

**Comment by bwims** [comment]
Thanks - I've already returned it.  I found that everything else had to be done in whole degree steps as well.... but 1 degree C is a massive amount of heat difference!

The main reason I have my ugly sonoff basic + BME280 on the wall is so that I had complete control over it via Node Red.  These restrictions by Moes are ridiculous!

Oh well!

**Reply by mhu1987** [reply]
Any update on this issue?

**Comment by PulsarFX** [comment]
@mhu1987 @bwims @Minims @kasi0815 @presslab-us   see #19991 if this is still an issue
